java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -1021,7 +1074,19 @@ private void setNullAt(int index) {
     }
 
     public final ColumnBuilder appendNull() {
-      growBuffersAndRows(true, 0);
+      // Increments row number. And update offsets and data buffer along with the valid buffer.
+      // NOTE: The growth of valid buffer must happen after rowCapacity updated.
+      if (type.typeId == DType.DTypeEnum.LIST) {
+        growListBuffersAndRows();
+      } else if (type.typeId == DType.DTypeEnum.STRING) {
+        growStringBuffersAndRows(0);
+      } else if (type.getSizeInBytes() > 0) {
+        growFixedWidthBuffersAndRows();
+      } else {
+        growStructBuffersAndRows();
+      }
+      growValidBuffer();
```

Comment:
This is called every time we append a null unconditionally.  It also allocates and copied the validity every time unconditionally, even if the buffer is large enough to not need to be re-allocated.  This feels strictly worse than the current implementation in this respect if there are a lot of nulls being entered.